blocks/api/serializer.js

```diff
@@ -36,35 +36,94 @@ export function getSaveContent( save, attributes ) {
 	return wp.element.renderToString( rawContent );
 }
 
+const escapeDoubleQuotes = value => value.replace( /"/g, '\"' );
+const replaceHyphens = value => value.replace( /-/g, '\\-' );
+
+/**
+ * Transform value for storage in block comment
+ *
+ * Some special characters and sequences should not
+ * appear in a block comment header. This transformer
+ * will guarantee that we store the data safely.
+ *
+ * @param {*}   value attribute value to serialize
+ * @returns {*}       transformed value
+ */
+const serializeValue = value =>
+	'string' === typeof value
```

Comment:
they shouldn't be. we should already be JSON-serializing data if not a number or string.

this will change if we stop using `name="value"` pairings, which is being worked on in another PR at the moment (not yet published)